lib/syslog/writer.go

```diff
@@ -244,6 +300,50 @@ func (c bufferedConn) Close() error                { return c.conn.Close() }
 func (c bufferedConn) Flush() error                { return c.buf.Flush() }
 func (c bufferedConn) Write(b []byte) (int, error) { return c.buf.Write(b) }
 
+func cachedDialWriter(network string, address string, config *tls.Config, socksProxy string) (w io.Writer, err error) {
+	if w, ok := writerCache.get(network, address, socksProxy); ok {
+		return w, nil
+	}
+	w, err = syncedDialWriter(network, address, config, socksProxy)
+	if err != nil {
+		writerCache.set(network, address, socksProxy, w)
+	}
+	return w, err
+}
+
+type dialerMutexMap struct {
+	sync.Mutex
+	mu map[string]*sync.Mutex
+}
+
+func (mm dialerMutexMap) lock(network string, address string, socksProxy string) {
+	key := writerCacheKey(network, address, socksProxy)
+	mm.Lock()
+	defer mm.Unlock()
+	mu, ok := mm.mu[key]
+	if !ok {
+		mu = new(sync.Mutex)
+		mm.mu[key] = mu
```

Comment:
The `mm.mu` map isn't initialized anywhere, this code will panic in this assignment.

Reply:
See below at line 341